engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala

```diff
@@ -0,0 +1,63 @@
+package pl.touk.nussknacker.engine.modelconfig
+
+import com.typesafe.config.{Config, ConfigFactory}
+import pl.touk.nussknacker.engine.modelconfig.ModelConfigLoader.defaultModelConfigResource
+
+object ModelConfigLoader {
+
+  private val defaultModelConfigResource = "defaultModelConfig.conf"
+
+}
+
+/**
+  * Loads model configs.
+  *
+  * Configs are resolved once when loading model data. Implement this trait when you want to add dynamic configuration
+  * that is fetched from external source, for example OpenAPI specification and you would like this configuration to
+  * be immutable for deployed processes.
+  *
+  * This trait is optional to implement.
+  */
+abstract class ModelConfigLoader extends Serializable {
+
+  /**
+    * Resolves config part to pass in execution during e.g. process deployment on Flink cluster. When running
+    * a process (e.g. on Flink) resolved config part is used to construct full config.
+    *
+    * Method used for performance reasons to reduce serialized configuration size inside deployed processes. By default
+    * config from main nussknacker file at path: processTypes.{type_name}.modelConfig is used.
+    *
+    * @param configDuringModelAnalysis default config resolved by [[resolveConfigDuringExecution]] for configuration
+    *                                  from processTypes.{type_name}.modelConfig
+    * @return transformation of config part (processTypes.{type_name}.modelConfig) that is later passed to a running
+    *         process (see e.g. FlinkProcessCompiler, ModelConfigToLoad).
+    */
+  def resolveConfigToPassInExecution(configDuringModelAnalysis: Config, classLoader: ClassLoader): ConfigToPassInExecution
```

Comment:
Why this method return class with `transform` instead just return result of `transform` or eagerly wrapped result of transformation? I can't read it from documentation :)

Reply:
The main reason is to distinguish which config can be modified since `configDuringModelAnalysis` should be read only and the one passed inside `transform` can be only changed.

Reply:
But typesafe config is immutable....

Reply:
`ConfigToPassInExecution` has been replaced with a value class that is used in the second step to resolve final config.